File: easybuild-backend/Libraries/Easybuild.DAL/Models/AdvertPageRequest.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Easybuild.DAL.Models
{
    public class AdvertPageRequest : PageRequest
    {
        public AdvertSortingOption AdvertSortingOption { get; set; }

        [StringLength(256)]
        public string SearchKeywords { get; set; }

        public bool SearchInDescription { get; set; }

        public List<int> JobCategoriesId { get; set; }

        public Role? Role { get; set; }

        public int UserId { get; set; }
    }
}

File: easybuild-frontend-web/src/app/shared/models/advert-page-request.ts
import { AdvertSortingOption, UserRole } from './enums';
import { PageRequest } from './page-request';

export class AdvertPageRequest extends PageRequest {
    advertSortingOption: AdvertSortingOption;
    searchKeywords: string;
    searchInDescription: boolean;
    jobCategoriesId?: number[];
    role?: UserRole;
    userId?: number;
}